back/engines/free/email_campaigns/spec/models/email_campaigns/campaigns/event_registration_confirmation_spec.rb

```diff
@@ -43,5 +43,21 @@
         image_url: event.project.header_bg.url
       })
     end
+
+    context 'when event has location' do
+      before { event.update!(location_point: 'POINT (1.0 2.0)') }
+
+      it 'can be processed by ActiveJob', :active_job_que_adapter do
+        stub_const 'TestJob', (Class.new(ApplicationJob) do
```

Comment:
Why are we using a test class instead of `EventRegistrationConfirmation` here?

Reply:
We need a job class here. We could use `TriggerOnActivityJob` though.

Reply:
Actually, it's used in the job of this mailer `EventRegistrationConfirmationMailer`. But it turned out it is much harder to set up than `TestJob`. So, I hope leaving it as it is is good enough for you.

Reply:
Yes, fine by me, but isn't there something like `:deliver_later`?

Reply:
Yes, there is, but then it complained that `recipient` was not configured and I gave up
```
     Failure/Error: email_address_with_name(recipient.email, "#{recipient.first_name} #{recipient.last_name}")

     NoMethodError:
       undefined method `email' for nil
```

Reply:
Hehe, alright let's forget about it.